dockstore-client/src/main/java/io/dockstore/client/cli/nested/ApiClientExtended.java

```diff
@@ -109,4 +137,187 @@ public void createBodyPart(MultiPart multiPart, String key, Object formObject) {
+            mergedHeaderMap.remove("Expect");
+        }
+
+        return mergedHeaderMap;
+    }
+
+    // TODO: Handle requests with body content.
+
+    /**
+     * Creates an Invocation.Builder that will be used to make a WES request. If the request is to be sent to an AWS endpoint
+     * a SigV4 Authorization header needs to be calculated based on the canonical request (https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html).
+     * @param requiresAwsHeaders Boolean value indicating if this request requires AWS-specific headers
+     * @param target The target endpoint
+     * @param method The HTTP method (GET, POST, etc ...)
+     * @param headerParams A list of header parameters custom to this request
+     * @return An Invocation.Builder that can be used to make an HTTP request
+     */
+    private Invocation.Builder createInvocation(boolean requiresAwsHeaders, WebTarget target, String method, Map<String, String> headerParams) {
+
+        // Merge all our different headers into a single object for easier handling
+        Map<String, String> mergedHeaderMap = mergeHeaders(requiresAwsHeaders, headerParams);
+
+        Invocation.Builder invocationBuilder = target.request();
+
+        // If the request is to an AWS endpoint, we need to take some extra steps to calculate a SigV4 Authorization header.
+        if (requiresAwsHeaders) {
+            HttpRequest request = new HttpRequest(method, target.getUri());
+            Signer.Builder awsAuthSignature = Signer.builder()
+                // TODO this currently only supports permanent AWS credentials
+                .awsCredentials(new AwsCredentials(this.wesRequestData.getAwsAccessKey(), this.wesRequestData.getAwsSecretKey()))
+                .region(this.wesRequestData.getAwsRegion())
+                .header(HttpHeaders.HOST, target.getUri().getHost()); // Have to manually set the Host header as it's required when signing
+
+            // add all the non-Authorization headers to both request objects simultaneously
+            for (Map.Entry<String, String> mapEntry : mergedHeaderMap.entrySet()) {
+                invocationBuilder.header(mapEntry.getKey(), mapEntry.getValue());
+                awsAuthSignature.header(mapEntry.getKey(), mapEntry.getValue());
+            }
+
+            // TODO: Remove this once we are making requests with a body.
+            String contentSha256 = org.apache.commons.codec.digest.DigestUtils.sha256Hex("");
+            String signature = awsAuthSignature.build(request, AWS_WES_SERVICE_NAME, contentSha256).getSignature();
+
+            // Add the new Authorization header SigV4 signature to the invocation
+            invocationBuilder.header(HttpHeaders.AUTHORIZATION, signature);
+
+        } else {
+            // add all the non-Authorization headers to the invocation object
+            for (Map.Entry<String, String> mapEntry : mergedHeaderMap.entrySet()) {
```

Comment:
Really nit-picky and opinionated, but I would move this loop up before the if statement, and just do it once. It looks like you are trying to save a second loop through the headers for the AWS code, but the object is so small I don't think it's worth it. Then you could simplify the method to be:

```
...
Invocation.Builder invocationBuilder = target.request();
for (Map.Entry<String, String> mapEntry : mergedHeaderMap.entrySet()) {
          invocationBuilder.header(mapEntry.getKey(), mapEntry.getValue());
}
final String authHeader = requiresAwsHeaders ? generateAwsSignature(...) : this.wesRequestData.getBearerToken();
invocationBuilder.header(HttpHeaders.AUTHORIZATION, authHeader);
return invocationBuilder;
```

You'd have to create the `generateAwsSignature()` method with the necessary parameters.

Personal preference, feel free to ignore if you like what you have.